adapters/repos/db/vector/cuvs/index.go

```diff
@@ -0,0 +1,739 @@
+		if id > maxID {
+			maxID = id
+		}
+	}
+
+	if vectors == nil || len(vectors) == 0 {
+		return
+	}
+
+	index.AddBatch(context.Background(), ids, vectors)
+}
+
+func (index *cuvs_index) Dump(labels ...string) {
+	if len(labels) > 0 {
+		fmt.Printf("--------------------------------------------------\n")
+		fmt.Printf("--  %s\n", strings.Join(labels, ", "))
+	}
+	fmt.Printf("--------------------------------------------------\n")
+	fmt.Printf("ID: %s\n", index.id)
+	fmt.Printf("--------------------------------------------------\n")
+}
+
+// searchbyvectordistance
+func (index *cuvs_index) SearchByVectorDistance(ctx context.Context, vector []float32, dist float32, maxLimit int64, allowList helpers.AllowList) ([]uint64, []float32, error) {
+	return []uint64{}, []float32{}, nil
+}
+
+func (index *cuvs_index) maxLimit() int64 {
+	return 1_000_000_000
+}
+
+func (index *cuvs_index) UpdateUserConfig(updated schemaConfig.VectorIndexConfig, callback func()) error {
+	return nil
+}
+
+func (index *cuvs_index) Drop(ctx context.Context) error {
+	if index.metadata != nil {
+		if err := index.metadata.Close(); err != nil {
+			return errors.Wrap(err, "close metadata")
+		}
+		if err := index.removeMetadataFile(); err != nil {
+			return err
+		}
+	}
+	// Shard::drop will take care of handling store's buckets
+	return nil
+}
+
+func CheckGpuMemory() error {
+	// run nvidia-smi
+	cmd := exec.Command("nvidia-smi")
+	out, err := cmd.Output()
+	println("nvidia-smi output (CheckGpuMemory()): ", string(out))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (index *cuvs_index) Stats() (common.IndexStats, error) {
+	return &CuvsStats{}, errors.New("Stats() is not implemented for cuvs index")
+}
+
+func (s *CuvsStats) IndexType() common.IndexType {
+	return common.IndexTypeFlat
+}
+
+type CuvsStats struct{}
+
+func (index *cuvs_index) Shutdown(ctx context.Context) error {
+	err := index.cuvs_internals.cuvsIndex.Close()
+	if err != nil {
+		return err
+	}
+
+	err = index.cuvs_internals.cuvsIndexParams.Close()
+	if err != nil {
+		return err
+	}
+
+	err = index.cuvs_internals.cuvsSearchParams.Close()
+	if err != nil {
+		return err
+	}
+
+	err = index.cuvs_internals.cuvsResource.Close()
+	if err != nil {
+		return err
+	}
+
+	err = index.cuvs_internals.dlpackTensor.Close()
+	if err != nil {
+		return err
+	}
+
+	if index.cuvsMemory != nil {
+		err = index.cuvsMemory.Close()
+		if err != nil {
+			return err
+		}
+	}
+
+	index.cuvs_internals.cuvsResource.Sync()
+
+	if index.metadata != nil {
+		if err := index.metadata.Close(); err != nil {
+			return errors.Wrap(err, "close metadata")
+		}
+	}
+
+	return nil
+}
+
+func (index *cuvs_index) Flush() error {
+	return nil
+}
+
+func (index *cuvs_index) SwitchCommitLogs(ctx context.Context) error {
+	return nil
+}
+
+func (index *cuvs_index) ListFiles(ctx context.Context, basePath string) ([]string, error) {
+	var files []string
+
+	if index.metadata != nil {
```

Comment:
Are we storing any cuVS files or always rebuilding on startup?

Reply:
we are always rebuilding on startup from the vectors on disk